[PT-D][Tensor Parallel] Add more test cases when we use use_orig_params for FSDP wrapping

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89779

Differential Revision: [D41600656](https://our.internmc.facebook.com/intern/diff/D41600656)

Commit:
[PT-D][Tensor Parallel] Sync change to pytorch and Add more test cases when we use use_orig_params for FSDP wrapping

[ghstack-poisoned]

wanchaol commented on torch/distributed/_tensor/ops/math_ops.py:
```diff
@@ -139,3 +140,35 @@ def var_correction_rule(op_schema: OpSchema) -> OutputSharding:
+
+
+@register_prop_rule("aten.native_layer_norm.default")
+def _prop_native_layer_norm(op_schema: OpSchema) -> OutputSharding:
```
can we delete this? This is actually right now an experimental op in https://github.com/pytorch/tau/blob/main/spmd/tensor/experimental_ops.py#L10

fduwjj replied:
Sure, totally.

wanchaol reviewed (approved):
looks good to me, one comment about removing experimental op

Commit:
Update on "[PT-D][Tensor Parallel] Sync change to pytorch and Add more test cases when we use use_orig_params for FSDP wrapping"




[ghstack-poisoned]

Commit:
Update on "[PT-D][Tensor Parallel] Sync change to pytorch and Add more test cases when we use use_orig_params for FSDP wrapping"




[ghstack-poisoned]

Commit:
Update on "[PT-D][Tensor Parallel] Sync change to pytorch and Add more test cases when we use use_orig_params for FSDP wrapping"




[ghstack-poisoned]

Commit:
Update on "[PT-D][Tensor Parallel] Sync change to pytorch and Add more test cases when we use use_orig_params for FSDP wrapping"




[ghstack-poisoned]

fduwjj commented:
@fduwjj has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41600656).

fduwjj commented:
@pytorchbot merge

fduwjj commented:
@pytorchbot rebase

Commit:
Update on "[PT-D][Tensor Parallel] Sync change to pytorch and Add more test cases when we use use_orig_params for FSDP wrapping"


Differential Revision: [D41600656](https://our.internmc.facebook.com/intern/diff/D41600656)

[ghstack-poisoned]

fduwjj commented:
@pytorchbot merge

fduwjj commented:
@pytorchbot merge

fduwjj commented:
@pytorchbot rebase

Commit:
Update on "[PT-D][Tensor Parallel] Add more test cases when we use use_orig_params for FSDP wrapping"


Differential Revision: [D41600656](https://our.internmc.facebook.com/intern/diff/D41600656)

[ghstack-poisoned]

fduwjj commented:
@fduwjj has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41600656).

fduwjj commented:
@pytorchbot merge
